compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -315,8 +320,8 @@ impl<Prov: Provenance, Bytes: AllocBytes> Allocation<Prov, (), Bytes> {
 
     /// Try to create an Allocation of `size` bytes, failing if there is not enough memory
     /// available to the compiler to do so.
-    pub fn try_uninit<'tcx>(size: Size, align: Align) -> InterpResult<'tcx, Self> {
-        Self::uninit_inner(size, align, || {
+    pub fn try_new<'tcx>(size: Size, align: Align, zero_init: bool) -> InterpResult<'tcx, Self> {
```

Comment:
Bools are a bit icky, as the call sites don't make it clear what the bool means. Maybe an enum with `Uninit` and `Zeroed` variants would be better? What do you think @RalfJung

Reply:
Yeah a dedicated `enum` makes sense.